### PR TITLE
[23.1] Fix up unit tests for local use

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes two little things when running unit tests locally

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: test/unit/app/managers/test_HistoryManager.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import time
+
```
Comment (nsoranzo) [inline test/unit/app/managers/test_HistoryManager.py]:
```suggestion
import time
```

Comment (mvdbeek) [reply test/unit/app/managers/test_HistoryManager.py]: Thanks, somehow I couldn't apply the suggestion, but I've included it in the rebase

Comment (nsoranzo) [reply test/unit/app/managers/test_HistoryManager.py]: No problem!

File: test/unit/app/queue_worker/conftest.py [hunk 2]
```diff
@@ -48,6 +48,9 @@ def create_app():
 def postgres_app(postgresql_proc):
     connection = "postgresql://{p.user}@{p.host}:{p.port}/".format(p=postgresql_proc)
 
+    if not psycopg2:
+        pytest.skip("psycopg2 must be installed for postgresql fixture")
```
Comment (nsoranzo) [inline test/unit/app/queue_worker/conftest.py]: From my quick test it looks like that, as per https://github.com/ClearcodeHQ/pytest-postgresql#how-to-use , it is actually psycopg (i.e. the [new version 3](https://github.com/psycopg/psycopg)) that needs to be installed for pytest-postgresql (in addition to psycopg2 for our SQLAlchemy connections).

In fact, that was the traceback I saw after installing `psycopg2-binary`:

```
$ pytest test/unit/app/queue_worker/test_database_heartbeat.py::test_database_heartbeat[postgres_app]
...
_________________________________________________________________________ ERROR at setup of test_database_heartbeat[postgres_app] _________________________________________________________________________

request = <SubRequest 'database_app' for <Function test_database_heartbeat[postgres_app]>>

    @pytest.fixture(params=["postgres_app", "sqlite_app", "sqlite_rabbitmq_app"])
    def database_app(request):
        if request.param == "postgres_app":
            if not which("initdb"):
                pytest.skip("initdb must be on PATH for postgresql fixture")
            if not psycopg2:
                pytest.skip("psycopg2 must be installed for postgresql fixture")
        if request.param == "sqlite_rabbitmq_app":
            if not os.environ.get("GALAXY_TEST_AMQP_INTERNAL_CONNECTION"):
                pytest.skip("rabbitmq tests will be skipped if GALAXY_TEST_AMQP_INTERNAL_CONNECTION env var is unset")
>       return request.getfixturevalue(request.param)

test/unit/app/queue_worker/conftest.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.10/site-packages/pytest_postgresql/factories/process.py:147: in postgresql_proc_fixture
    with DatabaseJanitor(
.venv/lib/python3.10/site-packages/pytest_postgresql/janitor.py:54: in __init__
    check_for_psycopg()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def check_for_psycopg() -> None:
        """
        Function checks whether psycopg was imported.
    
        Raises ImportError if not.
        """
        if not psycopg:
>           raise ImportError("No module named psycopg. Please install psycopg.")
E           ImportError: No module named psycopg. Please install psycopg.

.venv/lib/python3.10/site-packages/pytest_postgresql/compat.py:35: ImportError
```

I'll come up with a suggestion.

**Comment by nsoranzo** [comment]
I've pushed my suggested fix for the psycopg/psycopg2 test skipping, hope that's OK.

I've also pushed here a fix for unit/mypy test breakages due to the forward merge of https://github.com/galaxyproject/galaxy/pull/16480 .

**Pull request merged**